src/Pim/Bundle/CatalogBundle/Doctrine/ORM/Repository/ProductUniqueDataRepository.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\Doctrine\ORM\Repository;
+
+use Doctrine\ORM\EntityRepository;
+use Pim\Component\Catalog\Model\ProductInterface;
+use Pim\Component\Catalog\Model\ProductValueInterface;
+use Pim\Component\Catalog\Repository\ProductUniqueDataRepositoryInterface;
+
+/**
+ * Product unique data repository. Please see {@see Pim\Component\Catalog\Model\ProductUniqueDataInterface}
+ * for more information.
+ *
+ * @author    Julien Janvier <julien.janvier@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ProductUniqueDataRepository extends EntityRepository implements ProductUniqueDataRepositoryInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function uniqueDataNotExistsInThisProduct(ProductValueInterface $value, ProductInterface $product)
+    {
+        $queryBuilder = $this->createQueryBuilder('ud')
+            ->select('COUNT(ud)')
+            ->where('ud.attribute = :attribute')
+            ->andWhere('ud.rawData = :data')
+        ;
+
+        $parameters = [
+            'attribute' => $value->getAttribute(),
+            'data' => $value->__toString(),
```

Comment:
I double checked, it works well with supported attribute types: dates, identifiers, numbers and text.

I wonder if we should enforce that this work only with raw data, for instance if a partner tries to use it with my own attribute type (let's say option or reference data like) it would not work.